hescorehpxml/base.py

```
@@ -0,0 +1,2245 @@
+                bldg_about['town_house_walls'] = {'stand-alone': None,
+                                                  'attached on one side': 'back_right_front',
+                                                  'attached on two sides': 'back_front',
+                                                  'attached on three sides': None
+                                                  }[hpxml_surroundings]
+            except KeyError:
+                raise TranslationError('Site/Surroundings element is required in the HPXML document for town houses')
+            if bldg_about['town_house_walls'] is None:
+                raise TranslationError(
+                    'Cannot translate HPXML Site/Surroundings element value of %s into HEScore town_house_walls' %
+                    hpxml_surroundings)
+
+        bldg_cons_el = xpath(b, 'h:BuildingDetails/h:BuildingSummary/h:BuildingConstruction', raise_err=True)
+        bldg_about['year_built'] = int(xpath(bldg_cons_el, 'h:YearBuilt/text()', raise_err=True))
+        nbedrooms = int(xpath(bldg_cons_el, 'h:NumberofBedrooms/text()', raise_err=True))
+        if nbedrooms > 10:
+            nbedrooms = 10
+        bldg_about['number_bedrooms'] = nbedrooms
+        bldg_about['num_floor_above_grade'] = int(
+            math.ceil(float(xpath(bldg_cons_el, 'h:NumberofConditionedFloorsAboveGrade/text()', raise_err=True))))
+        avg_ceiling_ht = xpath(bldg_cons_el, 'h:AverageCeilingHeight/text()')
+        if avg_ceiling_ht is None:
+            try:
+                avg_ceiling_ht = float(xpath(bldg_cons_el, 'h:ConditionedBuildingVolume/text()', raise_err=True)) / \
+                                 float(xpath(bldg_cons_el, 'h:ConditionedFloorArea/text()', raise_err=True))
+            except ElementNotFoundError:
+                raise TranslationError(
+                    'Either AverageCeilingHeight or both ConditionedBuildingVolume and ConditionedFloorArea are required.'
+                    # noqa E501
```

If you put this on the end of a long line, this will silence the error about the line being too long.